blocks/news-feed/news-feed.js

```diff
@@ -0,0 +1,104 @@
+import ffetch from '../../scripts/ffetch.js';
+import { createOptimizedPicture, decorateIcons } from '../../scripts/lib-franklin.js';
+
+function buildPost(post, eager) {
+  const postCard = document.createElement('div');
+  postCard.classList.add('news-card');
+
+  const description = post.content.querySelector('h1 + p');
+  const title = post.content.querySelector('h1').textContent;
+
+  postCard.innerHTML = `
+      <div class="news-image">
+        <a href="${post.path}">${createOptimizedPicture(post.image, `Teaser image for ${title}`, eager).outerHTML}</a>
+      </div>
+      <div class="news-content">
+        <a class="post-title" href="${post.path}">${title}</a>
+        <a class="post-description" href="${post.path}">
+          <p>${description ? description.textContent : post.description}</p>
+          <span>Read More</span>
+        </a>
+      </div>
+    </a>
+  `;
+
+  return postCard;
+}
+
+async function buildNewsFeed(ul, pageNum, pageControl) {
+  const limit = 5;
+  const offset = pageNum * limit;
+  let morePages = false;
+  const blogPosts = ffetch('/query-index.json')
+    .filter((p) => p.path.startsWith('/news/'))
+    .slice(offset, offset + limit + 1)
+    .follow('path', 'content');
+
+  let i = 0;
+  const newUl = document.createElement('ul');
+  // eslint-disable-next-line no-restricted-syntax
+  for await (const post of blogPosts) {
+    if (i >= limit) {
+      // skip render, but know we have more page
+      morePages = true;
+      break;
+    }
+
+    const li = document.createElement('li');
+    li.append(buildPost(post, i < 1));
+    newUl.append(li);
+
+    i += 1;
+  }
+
+  pageControl.innerHTML = `
+      <ul class="pages">
+        <li class="prev"><a data-page="${pageNum - 1}" href="${window.location.pathname}?page=${pageNum}"><span class="icon icon-next"><span class="sr-only">Previous Page</span></a></li>
```

Comment:
I like that `data-page` is maintained here. Documenting this method to specify how `pageNum` is maintained would be great.

Reply:
added a comment which hopefully clarifies that